presto-spi/src/main/java/com/facebook/presto/spi/block/BlockBuilder.java

```diff
@@ -112,4 +112,10 @@ default BlockBuilder appendStructureInternal(Block block, int position)
      * Creates a new block builder of the same type based on the current usage statistics of this block builder.
      */
     BlockBuilder newBlockBuilderLike(BlockBuilderStatus blockBuilderStatus);
+
+    @Override
+    default int getOffsetBase()
+    {
+        return 0;
```

Comment:
nit: I personally prefer to have this method throw, and make the Block implementation to explicitly return 0 or other non-zero values.  -- This makes it easier to tell "this block is always a 0-offset block".

Reply:
Seems reasonable.